### ISSUE TITLE
Make the repository installable via a `setup.py` or `pyproject.toml`

### ISSUE THREAD

**Issue by seweber** [pairinteraction/pairinteraction#160]
We should make the repository installable via a `setup.py` or `pyproject.toml`. Then, it would be possible to easily install pairinteraction from the master branch:
```bash
pip install git+https://github.com/pairinteraction/pairinteraction
```

Moreover, using a  `pyproject.toml` together with [poetry](https://python-poetry.org/docs/) would solve #152. Potentially, it would also simplify using https://pypi.org/project/mkl as a dependency so that we do not have to deliver the MKL DLLs by our own. This would reduce the file size of the Windows wheel and solve #157.

To automatically call cmake within the Python build process, we could use [scikit-build-core](https://scikit-build-core.readthedocs.io/en/latest/index.html).

**Comment by hmenke** [comment]
It would probably be best to also move away from SWIG and switch to nanobind for this. My sparse matrix support for Eigen has already been merged some time ago.

**Comment by seweber** [comment]
I agree. I just found an example project that combines nanobind and scikit-build-core: https://github.com/wjakob/nanobind_example

**Comment by hmenke** [comment]
That's exactly what I was referring to. Migration to nanobind should also make it much easier to integrate pure Python extensions into pairinteraction.

**Comment by johannes-moegerle** [comment]
This issue should be resolved by the new version of pairinteraction.
If this issue continues to be unsolved for pairinteraction >= v1.4, please reopen this issue!
